src/main/java/org/opentripplanner/standalone/config/routerequest/RouteRequestConfig.java

```diff
@@ -462,6 +415,58 @@ private static void mapBikePreferences(NodeAdapter c, BikePreferences.Builder bu
             "How bad is it to walk the bicycle up/down a flight of stairs compared to taking a detour."
           )
           .asDouble(dft.stairsReluctance())
+      )
+      .withParking(it ->
+        it
+          .withUnpreferredVehicleParkingTagCost(
+            c
+              .of("unpreferredVehicleParkingTagCost")
+              .since(V2_3)
+              .summary("What cost to add if a parking facility doesn't contain a preferred tag.")
+              .description("See `preferredVehicleParkingTags`.")
+              .asInt(VehicleParkingPreferences.DEFAULT.unpreferredVehicleParkingTagCost().toSeconds())
+          )
+          .withBannedVehicleParkingTags(
+            c
+              .of("bannedVehicleParkingTags")
+              .since(V2_1)
+              .summary(
+                "Tags with which a vehicle parking will not be used. If empty, no tags are banned."
+              )
+              .asStringSet(List.of())
+          )
+          .withRequiredVehicleParkingTags(
+            c
+              .of("requiredVehicleParkingTags")
+              .since(V2_1)
+              .summary(
+                "Tags without which a vehicle parking will not be used. If empty, no tags are required."
+              )
+              .asStringSet(List.of())
+          )
```

Comment:
Should we remove these and instead create versions of these that are specific to car and bike?

Reply:
I think that is the correct way to do it. Remember that if you use the same type within each mode then you only need to write one config for it as well.

Q: `Tag` is very generic, it does not say witch context it comes from. Is it a `network`, a OSM  tag, or tag from somewhere else?

Reply:
I think I'll postpone the router-config refactoring to a follow up pr. Otherwise I would need to do it in two phases and this is a new feature anyway that no one is wanting to use yet.

The tag can originate from different places depending on the origin of the parking. I think mostly it's generated off some OSM tags but in Finland we load the tags from a custom format from a realtime updater. I think the network/feed is never a tag.

Reply:
Thank you for the explanation, a hint on this in the doc would be useful:

`The tag can originate from different places depending on the origin of the parking(OSM or RT feed).`